common/src/main/java/com/wynntils/features/chat/ChatTabsFeature.java

```diff
@@ -125,13 +128,38 @@ public void onScreenRender(ScreenRenderEvent event) {
 
     @SubscribeEvent
     public void onChatScreenKeyTyped(ChatScreenKeyTypedEvent event) {
-        // We can't use keybinds here to not conflict with TAB key's other behaviours.
-        if (event.getKeyCode() != GLFW.GLFW_KEY_TAB) return;
         if (!(McUtils.mc().screen instanceof ChatScreen)) return;
-        if (!KeyboardUtils.isShiftDown()) return;
 
-        event.setCanceled(true);
-        Services.ChatTab.setFocusedTab(Services.ChatTab.getNextFocusedTab());
+        int keyCode = event.getKeyCode();
+        if (keyCode == GLFW.GLFW_KEY_TAB) {
+            boolean reverse = false;
+
+            if (KeyboardUtils.isShiftDown()) {
+                if (oldTabHotkey.get()) {
+                    Services.ChatTab.setFocusedTab(Services.ChatTab.getNextFocusedTab());
+                    event.setCanceled(true);
+                    return;
+                } else {
+                    reverse = true;
+                }
+            }
+
+            if (KeyboardUtils.isControlDown() && !oldTabHotkey.get()) {
+                int newTab = reverse ? Services.ChatTab.getLastFocusedTab() : Services.ChatTab.getNextFocusedTab();
+                Services.ChatTab.setFocusedTab(newTab);
+                event.setCanceled(true);
+                return;
+            }
+        }
+
+        if (keyCode >= GLFW.GLFW_KEY_1 && keyCode <= GLFW.GLFW_KEY_9) {
+            ChatTab newTab = Services.ChatTab.getTab(keyCode - GLFW.GLFW_KEY_1);
+            if (newTab != null) {
+                Services.ChatTab.setFocusedTab(newTab);
+            }
+            event.setCanceled(true);
+            return;
+        }
```

Comment:
I can't see where this checks for CTRL being held down. In it's current form, it works by pressing the numbers only. Do I miss something?

Reply:
Well that's embarrassing, somehow I didn't notice I managed to miss this until now